cmd/lnd/main.go

```diff
@@ -14,8 +14,14 @@ func main() {
 	// function will also set up logging properly.
 	loadedConfig, err := lnd.LoadConfig()
 	if err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
+		if e, ok := err.(*flags.Error); err != nil &&
+			(!ok || e.Type != flags.ErrHelp) {
+			// Print error if not due to help request.
+			_, _ = fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		// Help was requested, exit normally.
```

Comment:
This is the default behavior. Also, the `err != nil` is already contained in the long condition, so you can really replace the whole `if err != nil ` with that if.

Reply:
Discussed this offline.

Reply:
Discard my previous comment, didn't look at the rest of the `main` function.